web/BaseApplication.php

```diff
@@ -87,41 +87,59 @@ public function resolveLanguageAndWebsite()
+                //Compare domain host. '//$domain' prevents parse_url failure, since parse_url requires url with schema
+                $domainHost = parse_url("//$domain", PHP_URL_HOST);
+                if ($domainHost == parse_url($this->request->getAbsoluteUrl(), PHP_URL_HOST)) {
+                    $domainPath = parse_url("//$domain", PHP_URL_PATH);
+                    $requestUrlParsed = parse_url($this->request->getAbsoluteUrl());
+                    $requestPath = $requestUrlParsed['path'];
+                    if (!$domainPath) {
+                        $domainPath = '/';
                     }
-                    $frontendHost = ArrayHelper::getValue($websiteData, 'frontendHost');
-                    if (!$frontendHost) {
-                        \Yii::warning("\"frontendHost\" does not exist for website \"$domain\"");
-                        $frontendHost = $domain;
+                    /* If no matching path found, the last domain is choosen,
+                       since the last domain has no path (after uksort() ordering)
+                    */
+                    if (preg_match("@^$domainPath/@", $requestPath . '/') || $domainPath == '/') {
+                        $this->websiteContentTree = ContentTree::findClean()
+                            ->byTableName(ContentTree::TABLE_NAME_WEBSITE)
+                            ->byKey($websiteKey)
+                            ->one();
+                        if (!$this->websiteContentTree) {
+                            throw new InvalidCallException("Current website does not exist");
+                        }
+                        $this->websiteMasterLanguage = $masterLanguage;
+                        $this->defaultContentId = $websiteData['defaultContentId'];
+                        $this->defaultContent = ContentTree::find()->byId($this->defaultContentId)->one();
+
+                        if (!$this->defaultContent) {
+                            throw new InvalidCallException("Default Content for website \"$websiteKey\" does not exist");
+                        }
+                        $this->language = $lang;
+                        $this->websiteLanguages = $this->getWebsiteLanguages($websiteKey);
+                        $shortCode = LanguageHelper::convertLongCodeIntoShort($this->language);
+                        if (strpos($domain, '/') !== false
+                            && (substr($domain, strpos($domain, '/') + 1) === $lang || $shortCode)) {
+                            $this->hasLanguageInUrl = true;
+                        }
+                        $frontendHost = ArrayHelper::getValue($websiteData, 'frontendHost');
+                        if (!$frontendHost) {
+                            \Yii::warning("\"frontendHost\" does not exist for website \"$domain\"");
+                            $frontendHost = $domain;
+                        }
+
+                        \Yii::setAlias('@frontendUrl', $requestUrlParsed['scheme'] . $frontendHost);
```

Comment:
need to add '://' after $requestUrlParsed['scheme'], since $requestUrlParsed['scheme'] contains only 'http' or 'https' without '://'